src/Oceananigans.jl

```diff
@@ -142,18 +138,36 @@ import Base:
     push!
 
 "Boolean denoting whether AMGX.jl can be loaded on machine."
-const hasamgx = @static Sys.islinux() ? true : false
+# TODO: find a way to check whether the libraries for AMGX and NETCDF 
+# (libamgxsh and libnetcdf, respectively) are installed on the machine

```

Comment:
this looks useful...

```Julia
julia> using Libdl

help?> find_library
search: find_library

  find_library(names, locations)


  Searches for the first library in names in the paths in the locations list, DL_LOAD_PATH, or system library paths (in that order) which can successfully be dlopen'd. On success, the return value will be one of
  the names (potentially prefixed by one of the paths in locations). This string can be assigned to a global const and used as the library name in future ccall's. On failure, it returns the empty string.
```

Reply:
nice

Reply:
I don't think it's useful since `find_library("libnetcdf")` gives empty for systems that NCDatasets works or not...

Reply:
there is some discussion that seems relevant at https://github.com/Alexander-Barth/NCDatasets.jl/issues/150